tool/ood-gen/lib/job.ml

```diff
@@ -0,0 +1,108 @@
+type metadata =
+  { title : string
+  ; link : string
+  ; description : string
+  ; location : string
+  ; company : string
+  ; company_logo : string
+  ; fullfilled : bool
+  }
+[@@deriving yaml]
+
+type t =
+  { id : int
+  ; title : string
+  ; link : string
+  ; description_html : string
+  ; location : string
+  ; company : string
+  ; company_logo : string
+  ; fullfilled : bool
+  }
+
+let path = Fpath.v "data/jobs.yml"
+
+let decode s =
+  let yaml = Utils.decode_or_raise Yaml.of_string s in
+  match yaml with
+  | `O [ ("jobs", `A xs) ] ->
+    Ok
+      (List.rev
+         (List.mapi
+            (fun i ->
+              Utils.decode_or_raise (fun x ->
+                  match metadata_of_yaml x with
+                  | Ok raw ->
+                    Ok
+                      { id = i
+                      ; title = raw.title
+                      ; link = raw.link
+                      ; description_html =
```

Comment:
I think a description is good, but if everyone is quite as thorough as you have been, this yaml file is going to get pretty big, pretty quickly. Do you think we should slim it down a bit (or alternative go for a collections model à la Tutorials)?

Reply:
I agree.. I didn't want to end up with a thousand files either. I was seeing two options:

- We don't include a description, so we can remove the opportunity detail page and simply redirect to the job post on the company page
- We can delete the job posts that are not available anymore, effectively keeping the file to a reasonable size.

Reply:
Yeah, I think trimming down the description is the way to go then. This will also lower the barrier for people adding their job postings (maybe at some point we could scrape some from discuss.ocaml.org). I think it would be nice to keep a record of jobs (I know they'll be there in git... but you know what I mean)

Reply:
I was thinking of it a bit more and it seems better to decide which implementation to go with once we've agreed on what we want to go on the website first.
From a product-only perspective, I'd be inclined to have a job description on the website. FWIW, this is what the Python job board looks like: https://www.python.org/jobs/
Thinking of data analytics, it could also be useful to have job descriptions.
What do you think? If we agree that we should only redirect to job posts on the companies website, let's trim down the descriptions as you suggested, but if we'd like to have job descriptions, I'm sure we can come up with a satisfying solution.